src/net/db.rs

```diff
@@ -1305,6 +1305,29 @@ impl PeerDB {
         PeerDB::get_random_neighbors(conn, network_id, count, block_height, true)
     }
 
+    pub fn get_valid_initial_neighbors(
+        conn: &DBConn,
+        network_id: u32,
+        block_height: u64,
+    ) -> Result<Vec<Neighbor>, db_error> {
+        // UTC time
+        let now_secs = util::get_epoch_time_secs();
+
+        let query = "SELECT * FROM frontier WHERE initial = 1 AND (allowed < 0 OR ?1 < allowed) \
+        AND network_id = ?2 AND denied < ?3 AND ?4 < expire_block_height"
```

Comment:
You should probably also filter by the `peer_version` major version byte -- i.e. `peer_version & 0xff000000`.  It should be less than or equal to that of the local peer's version.

This doesn't happen often, but when we have a breaking upgrade like Stacks 2.1, the major peer version byte will change, and for long-running nodes that will re-use the same chainstate, this means that their frontier table will have some peers that are still Stacks 2.0 peers.  Stacks 2.0 peers shouldn't try and talk to Stacks 2.1 peers, but Stacks 2.1 peers may be able to talk to Stacks 2.0 peers.